compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -561,38 +579,43 @@ private BType setLiteralValueAndGetType(BLangLiteral literalExpr, BType expType)
                     return setLiteralValueAndGetType(literalExpr, symTable.byteType);
                 }
 
-                finiteType = getFiniteTypeWithValuesOfSingleType((BUnionType) expType, symTable.byteType);
-                if (finiteType != symTable.semanticError) {
+                BType finiteTypeMatchWithByteType =
+                        getFiniteTypeWithValuesOfSingleType((BUnionType) expType, symTable.byteType);
+                if (finiteTypeMatchWithByteType != symTable.semanticError) {
+                    finiteType = finiteTypeMatchWithByteType;
                     BType setType = setLiteralValueAndGetType(literalExpr, finiteType);
                     if (literalExpr.isFiniteContext) {
                         // i.e., a match was found for a finite type
                         return setType;
                     }
                 }
 
-                if (memberTypes.stream().anyMatch(memType -> memType.tag == TypeTags.FLOAT)) {
-                    return setLiteralValueAndGetType(literalExpr, symTable.floatType);
-                }
-
-                finiteType = getFiniteTypeWithValuesOfSingleType((BUnionType) expType, symTable.floatType);
-                if (finiteType != symTable.semanticError) {
-                    BType setType = setLiteralValueAndGetType(literalExpr, finiteType);
-                    if (literalExpr.isFiniteContext) {
-                        // i.e., a match was found for a finite type
-                        return setType;
+                if (finiteType == symTable.semanticError) {
```

Comment:
Can we try to move out the common logic so that we can check int, byte, and so on in order?